armory/scenarios/video_ucf101_scenario_adversarial_preloaded_and_defended.py

```diff
@@ -0,0 +1,215 @@
+"""
+Classifier evaluation within ARMORY
+
+Scenario Contributor: MITRE Corporation
+"""
+
+import logging
+
+import numpy as np
+from tqdm import tqdm
+
+from armory.utils.config_loading import (
+    load_dataset,
+    load_model,
+    load_attack,
+    load_defense_wrapper,
+    load_defense_internal,
+    load,
+)
+from armory.utils import metrics
+from armory.scenarios.base import Scenario
+
+logger = logging.getLogger(__name__)
+
+
+class Ucf101(Scenario):
+    def _evaluate(self, config: dict) -> dict:
+        """
+        Evaluate the config and return a results dict
+        """
+
+        model_config = config["model"]
+        classifier, preprocessing_fn = load_model(model_config)
+
+        defense_config = config.get("defense") or {}
+        defense_type = defense_config.get("type")
+
+        IMAGE_DEFENSE_SUPPORT_ONLY_4D = True
+
+        if defense_type in ["Preprocessor", "Postprocessor"]:
+            if IMAGE_DEFENSE_SUPPORT_ONLY_4D:
+                # Architectures that use 3D convolution takes 5D
+                # inputs, usually in the shape of (nb, c, t, h, w)
+                # To use image-based defense that support 4D inputs,
+                # need to manually reshape 5D inputs into 4D, process
+                # defense, and then reshape back to 5D
+                defense = load(config.get("defense"))
+            else:
+                logger.info(f"Applying internal {defense_type} defense to classifier")
+                classifier = load_defense_internal(config["defense"], classifier)
```

Comment:
This hard coding makes it difficult for performers to apply to defenses that they add, since they would have to extend / or modify the scenario directly.

Reply:
I agree.  The performers could use this scenario only for baseline defense evaluation, and then use another scenario for evaluating their own defense.  Unless image-based defenses accepts 5D inputs, I'm not sure what cleaner workaround there is.